src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -149,16 +151,70 @@ private boolean validateDataSchema(Table table, Record record, boolean onlyField
+                    switch (colTypeName) {
+                        case "Date":
+                        case "Date32":
+                            if ( dataTypeName.equals(Type.INT32) || dataTypeName.equals(Type.STRING) ) {
+                                LOGGER.debug(String.format("Will try to convert from %s to %s", colTypeName, dataTypeName));
+                            }
+                            break;
+                        case "DateTime":
+                        case "DateTime64":
+                            if ( dataTypeName.equals(Type.INT64) || dataTypeName.equals(Type.STRING) ) {
+                                LOGGER.debug(String.format("Will try to convert from %s to %s", colTypeName, dataTypeName));
+                            }
+                            break;
+                        default:
+                            if (!colTypeName.equals(dataTypeName)) {
+                                validSchema = false;
+                                LOGGER.error(String.format("Table column name [%s] type [%s] is not matching data column type [%s]", col.getName(), colTypeName, dataTypeName));
+                            }
+
                     }
+
                 }
             }
         }
         return validSchema;
     }
 
+    private void doWriteDates(Type type, ClickHousePipedOutputStream stream, Data value) throws IOException {
+        boolean unsuported = false;
+        switch (type) {
+            case Date:
+                if (value.getFieldType().equals(Schema.Type.INT32)) {
+                    BinaryStreamUtils.writeUnsignedInt16(stream, ((Integer) value.getObject()).intValue());
+                } else {
+                    unsuported = true;
+                }
+                break;
+            case Date32:
+                if (value.getFieldType().equals(Schema.Type.INT32)) {
+                    BinaryStreamUtils.writeInt32(stream, ((Integer) value.getObject()).intValue());
+                } else {
+                    unsuported = true;
+                }
+                break;
+            case DateTime:
+                if (value.getFieldType().equals(Schema.Type.INT64)) {
+                    BinaryStreamUtils.writeUnsignedInt32(stream, ((Long) value.getObject()).longValue());
+                } else {
+                    unsuported = true;
+                }
+                break;
+            case DateTime64:
+                if (value.getFieldType().equals(Schema.Type.INT64)) {
+                    BinaryStreamUtils.writeInt64(stream, ((Long) value.getObject()).longValue());
+                } else {
+                    unsuported = true;
+                }
+                break;
+        }
+        if (unsuported) {
+            LOGGER.error("Not implemented conversion.");
+            throw new RuntimeException("Not implemented conversion.");
```

Comment:
There are no tests for the exception. We could move the `doWriteDates` into a separate file to simplify testing.